src/elisp/treemacs-mode.el

```diff
@@ -350,10 +350,15 @@ to it will instead show a blank."
       map)
     "Keymap for `treemacs-mode'."))
 
+(defvar treemacs--user-mode-line-format nil
```

Comment:
Since this is a user-facing customization variable it should be placed in treemacs-customization.el.

Being public it also shouldn't have a double-dash.

Finally it must be mentioned in the readme both in the [install snippet](https://github.com/Alexander-Miller/treemacs#installation) and the [config option table](https://github.com/Alexander-Miller/treemacs#variables).

Reply:
Thanks for the review (and sorry I overlooked the README.org update). I hope it's better now.

Reply:
LGTM. Squash it down to a single commit and we're good to merge.

Reply:
Done